actionpack/test/journey/route_test.rb

```diff
@@ -96,7 +96,7 @@ def test_score
         path = Path::Pattern.from_string "/:controller(/:action(/:id))(.:format)"
         generic = Route.build "name", nil, path, constraints, [], {}
 
-        knowledge = { id: 20, controller: "pages", action: "show" }
+        knowledge = { "id" => true, "controller" => true, "action" => true }
```

Comment:
I changed the signature of `score` to take a hash (acting as a set) of string keys.  `score` is called in a tight loop and previously had to convert the symbol keys to strings on every iteration.  I verified that `match_route` was the only place that this method was used internally, if there is fear that application code would use the scoring code, this can be reworked.